tests/Framework.py

```diff
@@ -171,25 +170,20 @@ def request(self, verb, url, input, headers):
 
     def __readNextRequest(self, verb, url, input, headers):
         fixAuthorizationHeader(headers)
-        self.__testCase.assertEqual(self.__protocol, readLine(self.__file))
-        self.__testCase.assertEqual(verb, readLine(self.__file))
-        self.__testCase.assertEqual(self.__host, readLine(self.__file))
-        self.__testCase.assertEqual(str(self.__port), readLine(self.__file))
-        self.__testCase.assertEqual(
-            self.__splitUrl(url), self.__splitUrl(readLine(self.__file))
-        )
-        self.__testCase.assertEqual(headers, eval(readLine(self.__file)))
+        assert self.__protocol == readLine(self.__file)
+        assert verb == readLine(self.__file)
+        assert self.__host == readLine(self.__file)
+        assert str(self.__port) == readLine(self.__file)
+        assert self.__splitUrl(url) == self.__splitUrl(readLine(self.__file))
+        assert headers == eval(readLine(self.__file))
         expectedInput = readLine(self.__file)
         if isinstance(input, str):
             if input.startswith("{"):
-                self.__testCase.assertEqual(
-                    json.loads(input.replace("\n", "").replace("\r", "")),
-                    json.loads(expectedInput),
-                )
+                assert json.loads(
+                    input.replace("\n", "").replace("\r", "")
+                ) == json.loads(expectedInput)
             else:
-                self.__testCase.assertEqual(
-                    input.replace("\n", "").replace("\r", ""), expectedInput
-                )
+                assert input.replace("\n", "").replace("\r", "") == expectedInput
```

Comment:
It seems we do the replace here no matter if the input is JSON or not, would you mind refactoring it so we do it unilaterally? Should also make the asserting for the json.loads() rather more readable too.

Reply:
Done.